lib/action_subscriber/publisher/async/in_memory_adapter.rb

```diff
@@ -62,6 +62,7 @@ class AsyncQueue
 
           def initialize
             @queue = ::Queue.new
+            @instrumentation_threadpool = ::Lifeguard::Threadpool.new
```

Comment:
I would prefer to see the `::ActiveSupport::Notifications.instrument` get called inline rather than put into a threadpool. We will end up using harness which is not really a concern for blocking the main thread (or the supervisor thread in this case).

If we don't need the threadpool then I would vote to remove it.